ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFFromUtcTimestamp.java

```diff
@@ -34,6 +35,7 @@
 import org.apache.hadoop.hive.serde2.objectinspector.primitive.PrimitiveObjectInspectorFactory;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import sun.util.calendar.ZoneInfo;
```

Comment:
I'm curious what are the advantages of this package. In my first impression, I would use `java.time.ZoneId` because it sounds more standard and consistent with the TimeZone API.

Reply:
TimeZone.getTimeZone uses ZoneInfo to get the proper timezone. But if the input is not a valid timezone, then it gives back the default one, and this is what this fix is about. Instead of the default it will throw exception to avoid data inconsistency.

Reply:
According to this : https://stackoverflow.com/a/55149856/7279976 
`sun.util.calendar.ZoneInfo; ` seems to be  replaced by [java.time.zone.ZoneRules](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/time/zone/ZoneRules.html) 

Can you please check using ZoneRules ?

Reply:
Refactored. Thanks for the input!